proposals/0001-gep.md

```diff
@@ -0,0 +1,198 @@
+---
+MEP: 1
+Title: Service Config
+Discussion: N/A
+Implementation: N/A
+---
+
+# Service Config
+
+## Abstract
+
+Glide is going to have many configs that control both functional and non-functional aspects of the service work. 
+This GEP defines how that config is going to be set, structured and maintained across the service lifetime.
+
+## Motivation
+
+Glide is a gateway that provides a bunch of configurations to set, manage, and maintain. 
+This ensures flexibility needed to meet various needs of our end users.
+
+For example:
+
+- **Telemetry Configs**: Log Output Format, Min Log Level, OTEL Configs
+- **HTTP Server Configs**: Is enabled?, IP and Port to bind, max body size, CORS, etc.
+- **SSL/TLS Configs:** Ciphers, Cert Paths, etc.
+- **LLM Provider Configs**: API Keys, Default Parameters, Timeouts, Connection Pool Configs, etc.
+- **Routing and Load Balancing Configs**: Balancing Strategy, Routing Weights, Provider Pool Definitions, etc.
+- **Caching Configs**: Redis Connection Configs, TTLs, etc.
+- **gRPC Server Configs**: Is enabled?, IP and Port to bind, etc.
+- etc.
+
+The configuration may be versioned via git to track and audit all changes. 
+At the same time, some configs are sensitive data (notably, LLM API Keys) 
+and it must not be provided directly in the config (otherwise, they would be leaked in git).
+
+As any other config, users should be able to easily apply new changes to the config. 
+The API keys may need to be rotated from time to time.
+
+Finally, the more configs we expose, the harder may become to understand the overall config
+and find the right setting to adjust. So the config structure is an important thing here to eliminate this fatigue.
+
+### Requirements
+
+- R1: Config format must be readable
+- R2: Config format must allow comments
+- R3: Config format must be `git diff`-able and versioned
+- R4: Config must be easy to understand and reason about
+- R5: Config must allow to include sensitive values from environment variables
+- R6: Config may allow to include sensitive values from other plain files
+- R7: Config should be easy to change and operate on
+
+## Design
```

Comment:
The minimum required config should be as limited as possible. For example, they should only need to provide 2 providers and the related apis keys. Everything else can be default.

Reply:
Yes, this was the strategy I wanted to pursue (not only with config, tbh, but also API schemas, and other contracts).  The full config is shown here to give a better idea on the proposed structure of the config. 

Theoretically, the pool config is the only thing that needs to be touched.

Reply:
Added an example of the most minimalistic config setup possible ✅